UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -364,26 +382,21 @@ public void SetReward(float reward)
         /// <param name="increment">Incremental reward value.</param>
         public void AddReward(float increment)
         {
-            m_Reward += increment;
-            m_CumulativeReward += increment;
+            Debug.Assert(DefaultRewardProvider != null, "the DefaultRewardProvider is null and " +
+                "method 'AddReward' was called.  If your agent doesn't have the CumulativeRewardProvider," +
+                "remove the call from 'AddReward'.");
+            DefaultRewardProvider.AddReward(increment);
         }
 
         /// <summary>
         /// Retrieves the step reward for the Agent.
         /// </summary>
         /// <returns>The step reward.</returns>
-        public float GetReward()
+        public float GetIncrementalReward()
         {
-            return m_Reward;
-        }
-
-        /// <summary>
-        /// Retrieves the episode reward for the Agent.
-        /// </summary>
-        /// <returns>The episode reward.</returns>
-        public float GetCumulativeReward()
-        {
-            return m_CumulativeReward;
+            Debug.Assert(rewardProvider != null, "m_RewardProviderComponent is null and " +
```

Comment:
nit: assert text doesn't match the others. 

Maybe just move all of these to something like
```csharp
void assertDefaultRewardProvider(string calllee) {
Debug.Assert(DefaultRewardProvider != null, "the DefaultRewardProvider is null and " +
  $"method '{callee}' was called.  If your agent doesn't have the CumulativeRewardProvider," +
  $"remove the call to '{callee}'.");
}
```
?

Reply:
(or ever better, AssertFormat)